src/internal/interfaces/Visualization.ts

```diff
@@ -99,18 +100,44 @@ export interface IRangeCondition {
 
 export type IMeasureValueFilterCondition = IComparisonCondition | IRangeCondition;
 
-export interface IBucketFilter {
-    allElements?: IBucketFilterElement[];
-    attribute?: string;
-    interval?: IBucketFilterInterval;
-    isInverted?: boolean;
-    isModified?: boolean;
-    noData?: boolean;
-    selectedElements?: IBucketFilterElement[];
-    totalElementsCount?: number;
-    overTimeComparisonType?: OverTimeComparisonType;
-    measureLocalIdentifier?: string;
-    condition?: IMeasureValueFilterCondition;
+export interface IAttributeFilter {
+    attribute: string;
+    isInverted: boolean;
+    totalElementsCount: number;
+    selectedElements: Array<{
+        title: string;
+        uri: string;
+    }>;
+}
+
+export interface IDateFilter {
+    attribute: "attr.datedataset";
+    overTimeComparisonType: "nothing" | "same_period_previous_year";
+    interval: {
+        granularity: string;
+        interval: [string, string] | [number, number];
+        name: string;
+        type: "relative" | "absolute";
+    };
+}
+
+export interface IMeasureValueFilter {
+    measureLocalIdentifier: string;
+    condition: IMeasureValueFilterCondition;
+}
+
+export type IBucketFilter = IAttributeFilter | IDateFilter | IMeasureValueFilter;
+
+export function isDateFilter(filter: IBucketFilter): filter is IDateFilter {
```

Comment:
These exported type guards are missing the unit tests.

Reply:
added